extract_desciptors_from_hpatch_file.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python2 -utt
+# -*- coding: utf-8 -*-
+import sys
+import numpy as np
+import lutorpy as lua
+import torchfile
+import os
+import sys
+import time
+#sys.path.insert(0, '/home/ubuntu/dev/opencv-3.1/build/lib')
+import cv2
+TFEAT_PATCH_SIZE = 32
+TFEAT_DESC_SIZE = 128
+TFEAT_BATCH_SIZE = 1000
+STATS_FNAME = 'nets/stats.liberty.t7'
+MODEL_FNAME = 'nets/tfeat_liberty_margin_star.t7'
+
+try:
+    stats = torchfile.load(STATS_FNAME)
+    MEAN = stats['mi']
+    STD = stats['sigma']
+except:
+    print "Please, do pip install torchfile, and ensure that there is nets/stats.liberty.t7 file"
+    sys.exit(1)
+
+def preprocess_patch(patch):
+    out = cv2.resize(patch, (TFEAT_PATCH_SIZE, TFEAT_PATCH_SIZE)).astype(np.float32) / 255;
+    out -= MEAN
+    out /= STD
+    return out.reshape(1, TFEAT_PATCH_SIZE, TFEAT_PATCH_SIZE)
+def extract_tfeats(net,patches):
+    num,channels,h,w = patches.shape
+    patches_t = torch.fromNumpyArray(patches)
+    patches_t._view(num, 1, TFEAT_PATCH_SIZE, TFEAT_PATCH_SIZE)
+    # split patches into batches
+    patches_t = patches_t._split(TFEAT_BATCH_SIZE)
+    descriptors = []
+    for i in range(int(np.ceil(float(num) / TFEAT_BATCH_SIZE))):
+        # infere Torch network
+        prediction_t = net._forward(patches_t[i]._cuda())
+        # Cast TorchTensor to NumpyArray and append to results
+        prediction = prediction_t.asNumpyArray()
+        # add the current prediction to the buffer
+        #print prediction.shape
```

Comment:
I will remove them, but they are taken from https://github.com/vbalnt/tfeat/blob/master/tfeat_demo.py#L97
If they are not needed, why do you included them?